app/gui2/src/stores/graph.ts

```diff
@@ -13,10 +13,12 @@ import {
   type IdMap,
   type NodeMetadata,
 } from 'shared/yjsModel'
-import { computed, reactive, ref, watch, watchEffect } from 'vue'
+import { computed, reactive, ref, watch, watchEffect, type Ref } from 'vue'
 import * as Y from 'yjs'
 import { useProjectStore } from './project'
 
+export type UnconnectedEdge = ({ source?: ExprId, target?: ExprId }) & { disconnectedEdgeTarget?: ExprId }
```

Comment:
if this *is* meant to be an intersection type, maybe prefer combining into a single object type.
alternatively, maybe something like this if it's meant to be a union instead:
```ts
export type UnconnectedEdge = 
| { source?: ExprId, target?: ExprId, disconnectedEdgeTarget?: never }
| { source?: never, target?: never, disconnectedEdgeTarget?: ExprId }
```